### PR TITLE
UCT/IB: complete TX ops with correct status

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
Complete TX ops with correct status

## Why ?
For following error handling improvements

### CODE REVIEW

**Comment by evgeny-leksikov** [comment]
bot:pipe:retest

File: src/uct/ib/dc/dc_mlx5.c [hunk 1]
```diff
@@ -205,12 +205,14 @@ uct_dc_mlx5_poll_tx(uct_dc_mlx5_iface_t *iface)
...
     uint16_t hw_ci;
     UCT_DC_MLX5_TXQP_DECL(txqp, txwq);
+    ucs_status_t status;
 
-    cqe = uct_ib_mlx5_poll_cq(&iface->super.super.super,
-                              &iface->super.cq[UCT_IB_DIR_TX]);
-    if (cqe == NULL) {
-        return 0;
+    status = uct_ib_mlx5_poll_cq(&iface->super.super.super,
+                              &iface->super.cq[UCT_IB_DIR_TX], &cqe);
```
Comment (brminich) [inline src/uct/ib/dc/dc_mlx5.c:211]: alignment

File: src/uct/ib/rc/accel/rc_mlx5.inl [hunk 2]
```diff
@@ -43,14 +43,14 @@ uct_rc_mlx5_common_update_tx_res(uct_rc_iface_t *rc_iface, uct_ib_mlx5_txwq_t *t
...
-                                uint16_t hw_ci)
+                                uint16_t hw_ci, ucs_status_t status)
 {
     if (cqe->op_own & MLX5_INLINE_SCATTER_32) {
         uct_rc_txqp_completion_inl_resp(txqp, cqe, hw_ci);
     } else if (cqe->op_own & MLX5_INLINE_SCATTER_64) {
         uct_rc_txqp_completion_inl_resp(txqp, cqe - 1, hw_ci);
     } else {
-        uct_rc_txqp_completion_desc(txqp, hw_ci);
+        uct_rc_txqp_completion_desc(txqp, hw_ci, status);
```
Comment (brminich) [inline src/uct/ib/rc/accel/rc_mlx5.inl:53]: would need to pass status to `uct_rc_txqp_completion_inl`_resp as well

**Comment by evgeny-leksikov** [comment]
bot:pipe:retest

**Comment by alinask** [comment]
can be part of https://github.com/openucx/ucx/pull/4743
closing for now.
@yosefe @evgeny-leksikov

**Pull request closed without merging**